File: java/spring-boot/src/main/java/com/backendtemplate/validation/ReservationValidation.java
package com.backendtemplate.validation;

import com.backendtemplate.exceptions.ReservationFailure;
import com.backendtemplate.exceptions.ReservationFailure.Reason;

public final class ReservationValidation {
    private ReservationValidation() {}

    public static void validateProductExists(boolean productExists) {
        if (!productExists) {
            throw new ReservationFailure(Reason.PRODUCT_NOT_FOUND);
        }
    }

    public static void validateReplayProduct(String requestedProductId, String replayProductId) {
        if (!requestedProductId.equals(replayProductId)) {
            throw new ReservationFailure(Reason.IDEMPOTENCY_CONFLICT);
        }
    }
}

File: kotlin/spring-boot/src/main/kotlin/com/backendtemplate/validation/ReservationValidation.kt
package com.backendtemplate.validation

import com.backendtemplate.exceptions.ReservationFailure
import com.backendtemplate.exceptions.ReservationFailure.Reason

fun validateProductExists(productExists: Boolean) {
    if (!productExists) throw ReservationFailure(Reason.PRODUCT_NOT_FOUND)
}

fun validateReplayProduct(requestedProductId: String, replayProductId: String) {
    if (requestedProductId != replayProductId) throw ReservationFailure(Reason.IDEMPOTENCY_CONFLICT)
}
